src/stats/overall.ts

```diff
@@ -97,7 +101,8 @@ function getOpeningRatio(conversionsByPlayerByOpening, playerIndex, opponentInde
   return getRatio(openings.length, openings.length + opponentOpenings.length);
 }
 
-function getBeneficialTradeRatio(conversionsByPlayerByOpening, playerIndex, opponentIndex) {
+// FIXME: Stop using any here. The type is actually one of those wacky custom objects generated by lodash
```

Comment:
I don't fully understand what this means. All lodash does is return standard js objects, what do you mean by wacky?

Reply:
Typescript is a superset of Javascript so it can't magically provide runtime type checking. It is only provides static type checking. Consider the following code:
```
const a = {
foo: 1,
bar: false,
baz: "hello world"
};
let str: string;
const x = _.get(a, [str]);
```
What type is x? How is your IDE meant to know what type `x` is so that it can warn you if you're accessing invalid object properties? It can't. You're right, Lodash does provide you a valid JS object, but as far as your IDE is concerned it has no idea what the type of `x` is which is what the `any` type is in Typescript. We can forgo all static type checking in code by saying they are of type `any` but it would best to avoid that since that defeats the purpose of type checking.

Reply:
I see, that makes sense, that said I don't practically think we do anything quite like that in the codebase. In the case below I think you're referring to the `conversionsByPlayerByOpening` correct? I imagine the type of that is actually something like:
```
{
  [playerIndex: number]: {
    [openingType: string]: ConversionType[]
  }
}
```

Reply:
Also with regards to a previous discussion, couldn't lodash types be specified on the variable they are initializing? In this case, for example:
```
const playerTrades: ConversionType[] = _.get(
  conversionsByPlayerByOpening, [playerIndex, 'trade']
 ) || [];
```

Though I do understand that it would be annoying to manually have to specify the type every time lodash it used.

Reply:
As mentioned in [this thread](https://github.com/project-slippi/slp-parser-js/pull/10#discussion_r319748853) you can get around the `any` by specifying a `_.Dictionary` type (one of Lodash's typings for the returned value of `_.get()`) but you'd have to do that every time `_.get()` is used. For now, I think it's better to just cast it as an `any`.